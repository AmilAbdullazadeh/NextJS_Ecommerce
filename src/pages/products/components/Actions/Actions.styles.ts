import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(({ spacing, breakpoints }) => ({
    button: {
        height: spacing(6),
        width: spacing(30),
        [breakpoints.up('md')]: {
            width: spacing(24),
        },
    },
}));

export default useStyles;
