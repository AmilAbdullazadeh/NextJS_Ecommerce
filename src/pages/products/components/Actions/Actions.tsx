import React from 'react';
import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';
import Box from '@material-ui/core/Box';
import { useTheme } from '@material-ui/core/styles';
import useMediaQuery from '@material-ui/core/useMediaQuery';

import { Typography } from '@components';

import useStyles from './Actions.styles';

export interface ActionProps {
    disabled: boolean;
    handleAddToCard: () => void;
    handleBuyNow: () => void;
}

const Actions: React.FC<ActionProps> = ({
    disabled,
    handleAddToCard,
    handleBuyNow,
}: ActionProps) => {
    const classes = useStyles();
    const theme = useTheme();
    const isLargeScreen = useMediaQuery(() => theme.breakpoints.up('md'));

    return (
        <Box py={8}>
            <Grid
                container
                direction={isLargeScreen ? 'row' : 'column'}
                justifyContent="center"
                alignItems="stretch"
                spacing={2}
            >
                <Grid item xs={12} md={6} lg={6} xl={6}>
                    <Button
                        data-testid="add-to-cart-btn"
                        variant="text"
                        style={{
                            backgroundColor: '#FFD814',
                        }}
                        className={classes.button}
                        onClick={handleAddToCard}
                        disabled={disabled}
                    >
                        <Typography variant="body1" component="span">
                            Add to Cart
                        </Typography>
                    </Button>
                </Grid>
                <Grid item xs={12} md={6} lg={6} xl={6}>
                    <Button
                        data-testid="buy-now-btn"
                        variant="text"
                        style={{
                            backgroundColor: '#FFA41C',
                        }}
                        className={classes.button}
                        onClick={handleBuyNow}
                        disabled={disabled}
                    >
                        <Typography variant="body1" component="span">
                            Buy Now
                        </Typography>
                    </Button>
                </Grid>
            </Grid>
        </Box>
    );
};

export default Actions;
