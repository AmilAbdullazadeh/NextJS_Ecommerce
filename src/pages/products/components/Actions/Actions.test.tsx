import React from 'react';

import { renderWithProviders } from '@testUtils';

import Actions, { ActionProps } from './Actions';

describe('Product page actions', () => {
    const defaultProps: ActionProps = {
        disabled: false,
        handleAddToCard: jest.fn(),
        handleBuyNow: jest.fn(),
    };

    it('render correctly', () => {
        const { container } = renderWithProviders(
            <Actions {...defaultProps} />
        );

        expect(container.firstChild).toMatchSnapshot();
    });

    it('render correctly with buttons disabled', () => {
        const { container } = renderWithProviders(
            <Actions {...defaultProps} disabled={true} />
        );

        expect(container.firstChild).toMatchSnapshot();
    });

    it('call handleAddToCard when clicking the add to cart button', () => {
        const { getByTestId } = renderWithProviders(
            <Actions {...defaultProps} />
        );

        const addToCardBtn = getByTestId('add-to-cart-btn');

        addToCardBtn.click();

        expect(defaultProps.handleAddToCard).toHaveBeenCalled();
    });

    it('call handleBuyNow when clicking the buy now button', () => {
        const { getByTestId } = renderWithProviders(
            <Actions {...defaultProps} />
        );

        const buyNowBtn = getByTestId('buy-now-btn');

        buyNowBtn.click();

        expect(defaultProps.handleBuyNow).toHaveBeenCalled();
    });

    it('do not call handleAddToCard or handleBuyNow when their respective buttons are disabled', () => {
        const { getByTestId } = renderWithProviders(
            <Actions {...defaultProps} disabled={true} />
        );

        const addToCardBtn = getByTestId('add-to-cart-btn');
        const buyNowBtn = getByTestId('buy-now-btn');

        addToCardBtn.click();
        buyNowBtn.click();

        expect(defaultProps.handleAddToCard).not.toHaveBeenCalled();
        expect(defaultProps.handleBuyNow).not.toHaveBeenCalled();
    });
});
