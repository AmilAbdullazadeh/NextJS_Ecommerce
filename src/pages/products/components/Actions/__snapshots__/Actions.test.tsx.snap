// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product page actions render correctly 1`] = `
<div
  class="MuiBox-root MuiBox-root-2"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column MuiGrid-justify-content-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-6 MuiGrid-grid-xl-6"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-1"
        data-testid="add-to-cart-btn"
        style="background-color: rgb(255, 216, 20);"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <span
            class="MuiTypography-root MuiTypography-body1"
          >
            Add to Cart
          </span>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-6 MuiGrid-grid-xl-6"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-1"
        data-testid="buy-now-btn"
        style="background-color: rgb(255, 164, 28);"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <span
            class="MuiTypography-root MuiTypography-body1"
          >
            Buy Now
          </span>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Product page actions render correctly with buttons disabled 1`] = `
<div
  class="MuiBox-root MuiBox-root-4"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column MuiGrid-justify-content-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-6 MuiGrid-grid-xl-6"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-3 Mui-disabled Mui-disabled"
        data-testid="add-to-cart-btn"
        disabled=""
        style="background-color: rgb(255, 216, 20);"
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <span
            class="MuiTypography-root MuiTypography-body1"
          >
            Add to Cart
          </span>
        </span>
      </button>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-6 MuiGrid-grid-xl-6"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-3 Mui-disabled Mui-disabled"
        data-testid="buy-now-btn"
        disabled=""
        style="background-color: rgb(255, 164, 28);"
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <span
            class="MuiTypography-root MuiTypography-body1"
          >
            Buy Now
          </span>
        </span>
      </button>
    </div>
  </div>
</div>
`;
